include/swift/AST/ASTContext.h

```diff
@@ -683,6 +685,12 @@ class ASTContext final {
   /// The loader is owned by the AST context.
   ClangModuleLoader *getClangModuleLoader() const;
 
+  /// Retrieve the DWARF module loader for this ASTContext.
+  ///
+  /// If there is no Clang module loader, returns a null pointer.
+  /// The loader is owned by the AST context.
+  ClangModuleLoader *getDWARFModuleLoader() const;
```

Comment:
Can you get rid of this? The Clang importer is special because the Swift compiler has to use it for other things. It would definitely be bad if we had to start special-casing the DWARF loader throughout the compiler as well.

Reply:
The only use of this method is in ASTDemangler.cpp in `ASTBuilder::findForeignTypeDecl()`, which tries the DWARFImporter after the ClangImporter failed. We could changing `ASTContext::getClangModuleLoader()` to return a list of ClangModuleLoaders sorted by the order in which they should be queried. Would that be better?